src/test/unit/mixer_tricopter_unittest.cc

```diff
@@ -21,55 +21,174 @@
 #include <limits.h>
 
 extern "C" {
-  #include "platform.h"
-  #include "flight/mixer.h"
-  #include "flight/mixer_tricopter.h"
-  #include "io/beeper.h"
-  #include "io/rc_controls.h"
-  #include "config/runtime_config.h"
-  #include "debug.h"
-  #include "drivers/sensor.h"
-  #include "drivers/accgyro.h"
-  #include "common/axis.h"
-  #include "common/filter.h"
+#include "debug.h"
+#include "platform.h"
+
+#include "config/runtime_config.h"
+
+#include "common/axis.h"
+#include "common/filter.h"
+
+#include "drivers/sensor.h"
+#include "drivers/accgyro.h"
+
+#include "flight/mixer.h"
+#define MIXER_TRICOPTER_INTERNALS
+#include "flight/mixer_tricopter.h"
+
+#include "io/beeper.h"
+#include "io/rc_controls.h"
+
+servoParam_t servoConf;
+mixerConfig_t mixerConfig;
+tailTune_t tailTune;
+int16_t servo[MAX_SUPPORTED_SERVOS];
+
+void tailTuneModeThrustTorque(thrustTorque_t *pTT, const bool isThrottleHigh);
 }
 
 #include "unittest_macros.h"
 #include "gtest/gtest.h"
 
+class ThrustFactorCalculationTest: public ::testing::Test {
+    // We expect factor = 1 / tan(angle) (but adjusted for formats)
+    // Say we want mixerConfig.tri_tail_motor_thrustfactor to be 139, i.e. the factor should be 13.9
+    // angle = 1 / atan(factor), according to #25
+    // adjust to decidegrees and multiply by servoAvgAngle.numOf
+    // i.e. multiply by 3000, then round to integer
+    // so even if 12345 looks like an arbitrarily chosen number, it is the result of this calculation and corresponds to 4.115 degrees.
+    // Due to possible rounding effects we add a tolerance to the test
+protected:
+    virtual void SetUp() {
+        memset(&servoConf, 0, sizeof(servoConf));
+        servoConf.min = DEFAULT_SERVO_MIN;
+        servoConf.max = DEFAULT_SERVO_MAX;
+        servoConf.middle = DEFAULT_SERVO_MIDDLE;
+        servoConf.rate = 100;
+        servoConf.forwardFromChannel = CHANNEL_FORWARDING_DISABLED;
+
+        // give all servos a default command
+        for (uint8_t i = 0; i < MAX_SUPPORTED_SERVOS; i++) {
+            servo[i] = DEFAULT_SERVO_MIDDLE;
+        }
+
+        triInitMixer(&servoConf, &servo[5], &mixerConfig);
+        tailTune.mode = TT_MODE_THRUST_TORQUE;
+        tailTune.tt.state = TT_WAIT_FOR_DISARM;
+        tailTune.tt.servoAvgAngle.numOf = 300;
+    }
+};
 
-TEST(ATest, test) {
-    ASSERT_TRUE(false);
+TEST_F(ThrustFactorCalculationTest, 139) {
+    // given
+    tailTune.tt.servoAvgAngle.sum = 12345;
```

Comment:
@lkaino; First this was hard to believe but very good you noticed. I think you are (almost) right about where the 0 degrees is but the function produce the same data anyway. Look here: [Book1.pdf](https://github.com/lkaino/Triflight/files/247650/Book1.pdf)

This graph shows your function in blue. Data taken from the code itself using a temp test case iterating a lot of printf.
It seems it will produce the same result in the region 1.5-20 as in 91.5-110 degrees.
The black dashed lines are what the simpler implementation will do. Current one to the left and after (still pending) correction on the right.

(About the "almost" above: I don't think the position at 90 deg is exactly vertical for the servo. It should be where the thrust force is vertical. Almost the same but not exactly.)